pkg/chrootarchive/chroot_linux.go

```diff
@@ -54,7 +50,10 @@ func chroot(path string) (err error) {
 	}()
 
 	if err := syscall.PivotRoot(path, pivotDir); err != nil {
-		// If pivot fails, fall back to the normal chroot after cleaning up temp dir for pivot_root
+		// If pivot fails, fall back to the normal chroot after cleaning up temp dir and mount for pivot_root
+		if err := syscall.Unmount(path, syscall.MNT_DETACH); err != nil {
```

Comment:
Is `DETACH` necessary? I thought it's sorta async and you're going to do something with it right after.